### PR TITLE
Implement the health check service of the TSP

### PR DESCRIPTION
**Pull request by tahini**
Add the checkHealth method to request server's health status

Signed-off-by: Geneviève Bastien <gbastien@versatic.net>

### CODE REVIEW

**Comment by tahini** [comment]
Requires this patch for the trace server: https://git.eclipse.org/r/c/tracecompass.incubator/org.eclipse.tracecompass.incubator/+/174907

File: src/models/health.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Model for a server health status response
+ */
+export interface HealthStatus {
+    /**
+     * The server status
+     */
+    status: 'UP' | 'DOWN';
```
Comment (muddana-satish) [inline src/models/health.ts:8]: Trying to understand. Why not a simple boolean?

**Comment by MatthewKhouzam** [comment]
Maybe for later, add server metrics, like heap and whatnot. maybe it can be done via prometheus?

File: src/models/health.ts (hunk 1 above)
Comment (marechal-p) [reply src/models/health.ts:8]: Also, is it possible for the server to be down and still answer `DOWN` ?

Comment (tahini) [reply src/models/health.ts:8]: That's from the TSP. We may want to add more status eventually, like OVERLOADED!

And yes, a server that may offer many services could say, "no, I don't answer to trace queries right now". Not the current implementation, but some implementation may.

**Review by bhufmann — APPROVED** [review]
Looks good to me.

**Pull request merged**